website/docs/configuration/functions/templatefile.html.md

```diff
@@ -64,6 +64,59 @@ backend 10.0.0.2:8080
 
 ```
 
+### Generating JSON or YAML from a template
+
+If the string you want to generate will be in JSON or YAML syntax, it's
+often tricky and tedious to write a template that will generate valid JSON or
+YAML that will be interpreted correctly when using lots of individual
+interpolation sequences and directives.
+
+Instead, you can write a template that consists only of a single interpolated
+call to either [`jsonencode`](./jsonencode.html) or
+[`yamlencode`](./yamlencode.html), specifying the value to encode using
+[normal Terraform expression syntax](/docs/configuration/expressions.html)
+as in the following examples:
+
+```
+${jsonencode({
+  "backends": [for addr in ip_addrs : "${addr}:${port}"],
+})}
+```
+
+```
+${yamlencode({
+  "backends": [for addr in ip_addrs : "${addr}:${port}"],
+})}
+```
+
+Given the same input as the `backends.tmpl` example in the previous section,
+this will produce a valid JSON or YAML representation of the given data
+structure, without the need to manually handle escaping or delimiters.
+In this case, the repetition based on elements of `ip_addrs` is achieved by
```

Comment:
I've got one comment, that doesn't block merging if you disagree!

The placement of this sentence feels a bit off to me - I feel like it belongs in the previous paragraph, which sets up the example that uses for_each.

Reply:
Interesting... :thinking: 

My rationale for this structure was that I wanted to put the example as early as possible and most of the words _after_ it because I hoped that the example would catch the reader's eye and cause them to start reading.

I'm also making an optimistic assumption that users are likely to learn `for` expression syntax before they learn template control syntax and thus for many readers this is just reminding them of something they already know and giving them a link to go refresh their memories on the details if they need to, rather than crucial for understanding the main point here that we can use functions inside the templates.

However, I'm happy to adjust if you feel that these examples are likely to be confusing without that prior context. I think the way I would adjust it is to use a simpler example at first -- one that only uses a simple variable reference, for example, and then introduce the example with a `for` expression only after the main point is made.

Reply:
I finally found the words to what threw me off: the phrase "In this case" made me assume that the example would _follow_, as opposed to phrasing such as "in the previous example" or "in the example above" or whatnot. 

Your rationale makes sense, though, and as I said I'm not going to fight you on what boils down to my (sometimes odd) personal preference - merge away!

Reply:
Thanks for that extra info! I will edit it to say "In the above example" (or similar) instead.